cmd/tools_netem.go

```diff
@@ -123,7 +124,16 @@ func netemSetFn(_ *cobra.Command, _ []string) error {
 	}()
 
 	err = nodeNs.Do(func(_ ns.NetNS) error {
-		link, err := net.InterfaceByName(netemInterface)
+		netemIfLink, err := netlink.LinkByName(netemInterface)
+		if err != nil {
+			// Look for aliased interfaces as well
+			netemIfLink, err = netlink.LinkByAlias(netemInterface)
+			if err != nil {
+				return err
+			}
+		}
+		netemIfName := netemIfLink.Attrs().Name
+		link, err := net.InterfaceByName(netemIfName)
```

Comment:
What I propose we should do with interface names (in general) is:

1. if an interface name contains `/` we swap it with `-` and always set this as altname (while keeping the alias)
2. that way, we can always use `netlink.LinkByName(sanitizedInterfaceName)` when fetching a link, regardless if it is long or short. With a single API call.
3. This would eliminated the logic to fetch a link via alias or name.

we can use smth like this

```go
func (s *Socket) SanitizeName() {
    var sb strings.Builder // Efficient way to build strings
    sb.Grow(len(s.Name))   // Allocate enough memory to avoid reallocation

    for _, char := range s.Name {
        switch char {
        case '.', ' ', '_':
            sb.WriteRune('-') // Replace with '-'
        default:
            sb.WriteRune(char) // Keep the original character
        }
    }

    s.Name = sb.String()
}
```

Should happen here https://github.com/vista-/containerlab/blob/7cb1875ea02b0647f9471beb61255d8e563ca73b/links/link.go#L414

Reply:
Added!